appinventor/components/src/com/google/appinventor/components/runtime/ChatBot.java

```diff
@@ -650,5 +725,20 @@ private X509Certificate[] getSystemCertificates() {
       return new X509Certificate[0];
     }
   }
+
+  // Copied from ImageBot
+  private File getOutputFile() throws IOException {
+    String tempdir = FileUtil.resolveFileName(form, "", form.DefaultFileScope());
+    if (tempdir.startsWith("file://")) {
+      tempdir = tempdir.substring(7);
+    } else if (tempdir.startsWith("file:")) {
+      tempdir = tempdir.substring(5);
+    }
+    Log.d(LOG_TAG, "tempdir = " + tempdir);
+    File outFile = File.createTempFile("ImageBot", ".png", new File(tempdir));
```

Comment:
"ChatBot"?

Reply:
Done, part of moving getOutputFile to FileUtil. It now takes an input parameter which is the prefix to use.